core/src/main/java/io/questdb/cairo/sql/async/PageFrameSequence.java

```diff
@@ -539,15 +544,8 @@ private boolean dispatch() {
 
     private void initRecord(SqlExecutionCircuitBreaker executionContextCircuitBreaker) {
         if (localRecord == null) {
-            final SqlExecutionCircuitBreakerConfiguration sqlExecutionCircuitBreakerConfiguration = executionContextCircuitBreaker.getConfiguration();
             localRecord = new PageFrameMemoryRecord();
-            if (sqlExecutionCircuitBreakerConfiguration != null) {
-                circuitBreaker = new NetworkSqlExecutionCircuitBreaker(sqlExecutionCircuitBreakerConfiguration, MemoryTag.NATIVE_CB2);
-            } else if (executionContextCircuitBreaker instanceof AtomicBooleanCircuitBreaker) {
-                circuitBreaker = executionContextCircuitBreaker;
-            } else {
-                circuitBreaker = NetworkSqlExecutionCircuitBreaker.NOOP_CIRCUIT_BREAKER;
-            }
+            circuitBreaker.init(executionContextCircuitBreaker);
```

Comment:
It's probably fine to move this call outside of the if block, to L552. That's because the wrapper class calls `resetTimer()` on its network CB in case of subsequent page frame sequence initialization. WDYT?

Reply:
agree, this should be outside of the if statement.
changing it.

Reply:
...and we should probably set the `fd` inside `init()` to make sure it is always copied.

Reply:
Yes, it's a good idea.

Potentially, what the wrapper CB should be doing is calling `init()` on its appropriate internal CB (network or atomic). This init call would copy the CB state: fd in case of the network CB and `AtomicBoolean` in case of the atomic CB. This way we don't need to care about wrapped circuit breaker's thread safety as we'll always use an internal CB. WDYT?

Reply:
When it is not a network CB, the wrapper just references the same atomic boolean CB, so I think everything uses the same instance.

Re AtomicBooleanCircuitBreaker's thread-safety, maybe `fd` and `testCount` could be volatile. Probably `fd` does not matter much, because it should be always -1, so in practice it is just `testCount`.

Reply:
Volatile `testCount` may have a high cost as it's frequently incremented. As for `fd` in `AtomicBooleanCircuitBreaker` it's only used for diagnostics purposes.

Reply:
moved the call outside from the `if` statement, and introduced the `init(executionContextCircuitBreaker)` method on the `SqlExecutionCircuitBreaker` interface.